### PR TITLE
Add parts of pg_class

### PR DESCRIPTION
**Pull request by JLDLaughlin**
Part of: https://github.com/MaterializeInc/materialize/issues/4326

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4332)

### CODE REVIEW

**Comment by JLDLaughlin** [comment]
Oops - have to add `relnamespace` to be useful for [`pg_catalog.Pg_table_is_visible`](https://github.com/MaterializeInc/materialize/issues/4282). Working on it!

File: src/coord/src/catalog/builtin.rs [hunk 1]
```diff
@@ -563,6 +563,34 @@ FROM mz_catalog.mz_schemas",
...
+    FROM mz_catalog.mz_sources
+    UNION SELECT oid, sinks AS relname
+    FROM mz_catalog.mz_sinks
+    UNION SELECT oid, views AS relname
+    FROM mz_catalog.mz_views
+    UNION (SELECT oid, name AS relname
+           FROM mz_catalog.mz_indexes
+           JOIN mz_catalog.mz_catalog_names
+           ON mz_catalog.mz_indexes.global_id = mz_catalog.mz_catalog_names.global_id
+           GROUP BY oid, name)
```
Comment (benesch) [inline src/coord/src/catalog/builtin.rs:588]: Seems like `mz_indexes` should grow a `name` column! This is from memory, so could be wrong, but pretty sure the name columns elsewhere are (correctly) not qualified, while the names in `mz_catalog_names` are!

**Review by benesch — APPROVED** [review]

File: src/coord/src/catalog/builtin.rs (hunk 1 above)
Comment (JLDLaughlin) [reply src/coord/src/catalog/builtin.rs:588]: yes, that's true! I think the complication has to do with what happens when the objects the index is on is renamed.. but I can look into it

**Comment by JLDLaughlin** [comment]
The join to support `relnamespace` is a bummer!

**Pull request merged**